client/src/components/Graph/Graph.tsx

```diff
@@ -1090,24 +1115,142 @@ class Graph extends React.Component<GraphProps, GraphState> {
     prevAsyncProps: GraphAsyncProps | null
   ): void {
     const {
-      positions,
+      positions: positionsEnd,
       colors,
       flags,
       height,
       width,
       imageUnderlay,
       screenCap,
+      layoutChoice,
+      layoutDf,
     } = asyncProps;
 
+    const positionsStart = prevAsyncProps?.positions ?? positionsEnd;
+    const prevLayoutChoice = prevAsyncProps?.layoutChoice;
+    this.duration =
+      layoutChoice.current !== prevLayoutChoice?.current &&
+      prevAsyncProps?.positions
+        ? globals.animationLength
+        : 0;
+
     this.cachedAsyncProps = asyncProps;
-    const { pointBuffer, colorBuffer, flagBuffer } = this.state;
+    const {
+      pointBuffer,
+      pointBufferStart,
+      pointBufferEnd,
+      pointBufferTerminal,
+      colorBuffer,
+      flagBuffer,
+    } = this.state;
     let needToRenderCanvas = false;
     if (height !== prevAsyncProps?.height || width !== prevAsyncProps?.width) {
       needToRenderCanvas = true;
     }
-    if (positions !== prevAsyncProps?.positions) {
+    if (positionsEnd !== prevAsyncProps?.positions) {
```

Comment:
add more in-line comments in this block

Reply:
Addressed: Added detailed inline comments explaining the four cases of position mapping logic, including what each case handles and why duration is set to 0 in certain scenarios.